src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Values must be greater than 0');
+    }
+
+    const p = this.a + this.b + this.c;
+    const maxValue = Math.max(this.a, this.b, this.c);
+
+    if (maxValue >= p - maxValue) {
```

Comment:
In the Triangle class, the variable `p` is calculated as the perimeter, but it should be the semi-perimeter (p/2) for the triangle inequality check. This is because the check `maxValue >= p - maxValue` should actually be `maxValue >= semip`, where `semip` is the semi-perimeter.